app/ts/components/Submit.tsx

```diff
@@ -5,240 +5,246 @@ import { Button } from './Button.js'
-				bundleStatus.value = { ...bundleStatus.value, pendingBundles }
-			}
-		}
-	}
-
-	async function toggleSubmission() {
-		if (!bundleStatus.peek().active) {
-			const relayProvider = flashbotsProvider.value ?? (await createProvider(provider))
-			if (!flashbotsProvider.value) flashbotsProvider.value = relayProvider
-			if (!provider.value) throw 'User not connected'
-			if (!interceptorPayload.value) throw 'No imported bundle found'
-			bundleStatus.value = {
-				active: true,
-				lastBlock: bundleStatus.value.lastBlock,
-				pendingBundles: bundleStatus.value.pendingBundles.filter((x) => x.state === 'pending'),
-			}
-			bundleSubmission(blockInfo.value.blockNumber)
-		} else {
-			bundleStatus.value = { ...bundleStatus.value, active: false }
-		}
-	}
-
-	return (
-		<>
-			<h2 className='font-bold text-2xl'>3. Submit</h2>
-			{missingRequirements.value ? (
-				<p>{missingRequirements.peek()}</p>
-			) : (
-				<div className='flex flex-col w-full gap-6'>
-					<Button onClick={simulateBundle} variant='secondary'>
-						Simulate
-					</Button>
-					<SimulationPromiseBlock state={simulationResult} />
-					<Button onClick={toggleSubmission}>{bundleStatus.value.active ? 'Stop' : 'Submit'}</Button>
-					<Bundles pendingBundles={bundleStatus} appSettings={appSettings} />
-				</div>
-			)}
-		</>
-	)
+  const [simulationResult, setSimulationResult] = useState<
+    | {
+      status: 'pending' | 'resolved' | 'rejected'
+      value?: SimulationResponseSuccess
+      error?: RelayResponseError
+    }
+    | undefined
+  >(undefined)
+
+  const flashbotsProvider = useSignal<FlashbotsBundleProvider | undefined>(undefined)
+
+  const bundleStatus = useSignal<{ lastBlock: bigint; active: boolean; pendingBundles: BundleInfo[] }>({
+    active: false,
+    lastBlock: blockInfo.peek().blockNumber,
+    pendingBundles: [],
+  })
+
+  useSignalEffect(() => {
+    blockInfo.value.blockNumber // trigger effect
+    if (bundleStatus.peek().active && blockInfo.value.blockNumber > bundleStatus.peek().lastBlock) {
+      bundleSubmission(blockInfo.value.blockNumber)
+    }
+  })
+
+  const missingRequirements = useComputed(() => {
+    if (!interceptorPayload.value) return 'No transactions imported yet.'
+    const missingSigners = interceptorPayload.value.uniqueSigners.length !== Object.keys(signers.value.bundleSigners).length
+    const insufficientBalance = signers.value.burnerBalance < fundingAmountMin.value
+    if (missingSigners && insufficientBalance) return 'Missing private keys for signing accounts and funding wallet has insufficent balance.'
+    if (missingSigners) return 'Missing private keys for signing accounts.'
+    if (insufficientBalance) return 'Funding wallet has insufficent balance.'
+    return false
+  })
+
+  async function ensureRelayProvider() {
+    const relay = `${CORS_PROXY_PREFIX}${appSettings.peek().relayEndpoint}`
```

Comment:
Potentially do not need CORS_PROXY, or the one provided in `/ts/constants.ts` should be changed to something else.

Reply:
I think removing the CORS proxy for now, we have the goerli and mainnet endpoints that work fine. If someone uses a custom relay RPC for now they must provide one that works with CORS.

Reply:
Also was a potential external service that could go down and prevent use of the app.